File: GasEta/app/src/main/java/br/com/dnsistemas/gaseta/SplashActivity.java
package br.com.dnsistemas.gaseta;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

public class SplashActivity extends AppCompatActivity {

    public static final int TIME_OUT_SPLASH = 3000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        comutarTelaSplash();
    }

    private void comutarTelaSplash() {

        new Handler().postDelayed(() -> {
            Intent telaPrincipal = new Intent(SplashActivity.this, MainActivity.class);
            startActivity(telaPrincipal);
            finish();
        }, TIME_OUT_SPLASH);
    }
}

File: AppBaseKotlin/app/src/main/java/br/com/dnsistemas/appbasekotlin/view/SplashActivity.kt
package br.com.dnsistemas.appbasekotlin.view

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import br.com.dnsistemas.appbasekotlin.R
import br.com.dnsistemas.appbasekotlin.databinding.ActivitySplashBinding
import br.com.dnsistemas.appbasekotlin.util.Constantes

class SplashActivity : AppCompatActivity() {

    private lateinit var binding: ActivitySplashBinding

    //TODO: Verificar se existe o sharedpreferences caso não exista criar e direcionar para a tela de login

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySplashBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.hide()

        if (isLogued()){
            comutarTela(Intent(this, HomeActivity::class.java))
        }else {
            comutarTela(Intent(this, MainActivity::class.java))
        }

    }

    private fun isLogued(): Boolean {
        return getPreferences(MODE_PRIVATE).getBoolean("logued", false)
    }


    fun comutarTela(intent: Intent){
       Handler(Looper.getMainLooper()).postDelayed({
           startActivity(intent)
           finish()
       }, 3000)
   }
}
